compiler/damlc/tests/daml-test-files/InterfaceGuardedNotExtended.daml

```diff
@@ -63,56 +63,59 @@ exception GuardException
     where
         message m
 
+assetAsTokenTpl : Party -> Token
+assetAsTokenTpl p =
```

Comment:
Keeping in mind how we'll tell people to migrate minimally, given

```hs
template T with x : Text, y : Text
  ...
  let thing = "Value of x is " <> x
  choice C : Text
    ...
    do pure thing
```

do we want to tell them to turn that into:

```
f this@T{..} = "Value of x is " <> x
...
  choice C : Text
    ...
    do pure (thing this)
```

or do we want to tell them to bind only the args they need

```
f x = "Value of x is " <> x
...
  choice C : Text
    ...
    do pure (thing x)
```

I think we can use either approach here because it's a test, but I'd lean towards the former for simplicity's sake in docs and other user facing code.

Reply:
I agree with using the former, adding a couple of cleanup steps (remove `this` if unused, replace `..` with actual fields needed), which is actually what I ended up doing in https://github.com/digital-asset/daml-finance/pull/997/files